rotkehlchen/chain/evm/decoding/hop/balances.py

```diff
@@ -0,0 +1,150 @@
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
+        """Queries and returns the balance sheet for staking events.
+
+        This method:
+        1. Retrieves deposit events for staking.
+        2. Groups addresses by staking contract.
+        3. For each contract, queries user balances and rewards.
+        4. Converts balances and rewards to USD and updates the balance sheet."""
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+        if len(addresses_with_deposits := self.addresses_with_deposits(products=[EvmProduct.STAKING])) == 0:  # noqa: E501
+            return balances
+
+        # Group addresses by staking contract
+        staked_contracts: dict[ChecksumEvmAddress, set[ChecksumEvmAddress]] = defaultdict(set)
+        for user_address, events in addresses_with_deposits.items():
+            for event in events:
+                if event.address:
+                    staked_contracts[event.address].add(user_address)
+
+        for contract_address in staked_contracts:
+            addresses = staked_contracts[contract_address]
+            staking_contract = EvmContract(
+                address=contract_address,
+                abi=self.evm_inquirer.contracts.abi('HOP_STAKING'),
+                deployed_block=0,
+            )
+            chunk_size, call_order = get_chunk_size_call_order(self.evm_inquirer)
+
+            if len(staked_lps := self.evm_inquirer.multicall(
+                calls=[(
+                    staking_contract.address,
+                    staking_contract.encode(
+                        method_name='balanceOf',
+                        arguments=[user],
+                    ),
+                ) for user in addresses],
+                call_order=call_order,
+                calls_chunk_size=chunk_size,
+            )) == 0:
+                return balances
+
+            token_data = self.evm_inquirer.multicall(
+                calls=[
+                    (staking_contract.address, staking_contract.encode(method_name='stakingToken')),  # noqa: E501
+                    (staking_contract.address, staking_contract.encode(method_name='rewardsToken')),  # noqa: E501
+                ],
+                call_order=call_order,
+                calls_chunk_size=chunk_size,
+            )
+            checksummed_staking_token, checksummed_rewards_token = to_checksum_address(
+                staking_contract.decode(result=token_data[0], method_name='stakingToken')[0],
+            ), to_checksum_address(
+                staking_contract.decode(result=token_data[1], method_name='rewardsToken')[0],
+            )
+            staking_token, rewards_token = Asset(
+                evm_address_to_identifier(
+                    address=checksummed_staking_token,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            ).resolve_to_evm_token(), Asset(
```

Comment:
istead of resolving to evm_token you can use `EvmToken` instead of `Asset` and that will resolve. Also I believe you need to handle the UnknownAsset error here